controllers/controller.js

```diff
@@ -1,14 +1,25 @@
-const { sendTopics } = require("../models/model");
+const { sendTopics, selectArticleById } = require("../models/model");
 const endpoints = require("../endpoints.json");
 
 exports.getEndpoints = (req, res) => {
   res.status(200).send({ endpoints });
 };
 
 exports.getTopics = (req, res) => {
-  sendTopics()
-    .then((topic) => {
-      res.status(200).send({ topic });
+  sendTopics().then((topic) => {
+    res.status(200).send({ topic });
+  });
+};
+
+exports.getArticleByid = (req, res, next) => {
+  const { article_id } = req.params;
+  selectArticleById(article_id)
+    .then((article) => {
+      if (article !== undefined) {
+        res.status(200).send({ article });
+      } else {
+        res.status(404).send({ msg: "Article does not exist" });
+      }
```

Comment:
You should deal with this in the model instead of here and then you would be making use of your catch block below and error handlers.

Instead of checking if article is undefined, check the rows length in the model. If empty you will need to use Promise.reject to send an error to the catch block.

Then you will need a separate error handler to deal with custom errors. 

Once you set this up though, you will be able to use it again and again for other endpoints instead of repeating code like above for each controller.